tests/core/contracts/test_contract_call_interface.py

```diff
@@ -423,3 +424,23 @@ def test_returns_data_from_specified_block(web3, math_contract):
 
     assert output1 == 1
     assert output2 == 2
+
+
+def test_function_does_not_exist(arrays_contract):
+    with pytest.raises(AttributeError):
```

Comment:
[`pytest.raises`](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions) takes a `match` keyword which allows you to match against the exception message itself.  I suspect that we can do some very basic matching on the function names to be sure we're getting the *right* exception.